### ISSUE TITLE
tests/python: concurrency assertion fails under llvm-cov instrumentation

### ISSUE THREAD

**Issue by teng-lin** [teng-lin/rookie-cookies#337]
## Summary

`test_parallel_extractions_overlap_rather_than_queueing` (`tests/python/test_binding_runtime.py:313`) asserts wall-clock parallelism, but the `Python binding coverage` lane runs it against a **coverage-instrumented Rust build**. LLVM source-based coverage increments shared atomic counters on every basic block; four threads executing the same extraction path contend on those counter cache lines, which flattens parallel scaling to roughly 1×. The assertion then fails.

This is not flake in the usual sense — it is a deterministic interaction that trips whenever runner load pushes the ratio past the bound.

## Evidence

The binding genuinely parallelizes. Measured locally (uninstrumented, 10 cores, 4 workers):

```
serial baseline: 0.0738s
parallel(4):     0.1203s  vs 4*serial=0.2952s  speedup=2.45x  -> PASS
```

The GIL is correctly released — `bindings/python/src/browsers.rs:655` wraps `from_path` in `.detach()`.

Observed CI failures, both in `Python binding coverage (macos-latest)`:

| Run | Serial baseline | 4x serial | Parallel actual | Effective speedup |
|---|---|---|---|---|
| A | 0.530s | 2.120s | 2.201s | ~0.96x |
| B | 0.473s | 1.892s | 1.705s | ~1.11x |

Crucially, in the same workflow run, **all four plain `Python build and tests (macos-latest, 3.11/3.12/3.13/3.14)` jobs pass** while executing the same `tests/python` suite. The only differing variable is instrumentation: `scripts/run-python-coverage.py` uses `cargo llvm-cov show-env` to inject a `RUSTC_WRAPPER` and `RUSTFLAGS` so the binding is compiled with `-C instrument-coverage`.

## Proposed fix

Skip the assertion when coverage instrumentation is active — e.g. detect `LLVM_PROFILE_FILE` (already exported by `show-env`) and `skipTest`, or have `run-python-coverage.py` set an explicit marker env var the test honours.

**Do not simply loosen the bound.** The test exists to catch a binding that holds the GIL or funnels extractions through one mutex; a bound loose enough to pass under instrumentation would no longer detect real serialization.

## Cost of leaving it

Each occurrence costs a full `test-rust.yml` rerun during release gating. It surfaced twice while gating 0.6.0.
